tests/stdlib/tsugar.nim

```diff
@@ -26,85 +94,95 @@ block dup_with_field:
-doAssert b[0] == 0
-doAssert b[1] == 1
+# collect
 
-#test collect
-import sets, tables
+import std/[sets, tables]
 
 let data = @["bird", "word"] # if this gets stuck in your head, its not my fault
+
 doAssert collect(newSeq, for (i, d) in data.pairs: (if i mod 2 == 0: d)) == @["bird"]
 doAssert collect(initTable(2), for (i, d) in data.pairs: {i: d}) == {0: "bird",
       1: "word"}.toTable
 doAssert initHashSet.collect(for d in data.items: {d}) == data.toHashSet
 
-let x = collect(newSeqOfCap(4)):
+block:
+  let x = collect(newSeqOfCap(4)):
+      for (i, d) in data.pairs:
+        if i mod 2 == 0: d
+  doAssert x == @["bird"]
+
+block: # bug #12874
+  let bug = collect(
+      newSeq,
+      for (i, d) in data.pairs:(
+        block:
+          if i mod 2 == 0:
+            d
+          else:
+            d & d
+        )
+  )
+  doAssert bug == @["bird", "wordword"]
+
+import std/strutils
+
+block:
+  let y = collect(newSeq):
     for (i, d) in data.pairs:
-      if i mod 2 == 0: d
-doAssert x == @["bird"]
-
-# bug #12874
-
-let bug1 = collect(
-    newSeq,
-    for (i, d) in data.pairs:(
-      block:
-        if i mod 2 == 0:
-          d
-        else:
-          d & d
-      )
-)
-doAssert bug1 == @["bird", "wordword"]
-
-import strutils
-let y = collect(newSeq):
-  for (i, d) in data.pairs:
-    try: parseInt(d) except: 0
-doAssert y == @[0, 0]
-
-let z = collect(newSeq):
-  for (i, d) in data.pairs:
-    case d
-    of "bird": "word"
-    else: d
-doAssert z == @["word", "word"]
-
-proc tforum =
-  let ans = collect(newSeq):
-    for y in 0..10:
-      if y mod 5 == 2:
-        for x in 0..y:
-          x
-tforum()
+      try: parseInt(d) except: 0
+  doAssert y == @[0, 0]
+
+block:
+  let z = collect(newSeq):
+    for (i, d) in data.pairs:
+      case d
+      of "bird": "word"
+      else: d
+  doAssert z == @["word", "word"]
+
+block:
+  proc tforum(): seq[int] =
+    collect(newSeq):
+      for y in 0..10:
+        if y mod 5 == 2:
+          for x in 0..y:
+            x
+  doAssert tforum() == @[0, 1, 2, 0, 1, 2, 3, 4, 5, 6, 7]
 
 block:
   let x = collect:
     for d in data.items:
       when d is int: "word"
       else: d
   doAssert x == @["bird", "word"]
-doAssert collect(for (i, d) in pairs(data): (i, d)) == @[(0, "bird"), (1, "word")]
-doAssert collect(for d in data.items: (try: parseInt(d) except: 0)) == @[0, 0]
-doAssert collect(for (i, d) in pairs(data): {i: d}) == {1: "word",
-    0: "bird"}.toTable
-doAssert collect(for d in data.items: {d}) == data.toHashSet
-
-# bug #14332
-template foo =
-  discard collect(newSeq, for i in 1..3: i)
-foo()
\ No newline at end of file
+
+block:
+  doAssert collect(for (i, d) in pairs(data): (i, d)) == @[(0, "bird"), (1, "word")]
+  doAssert collect(for d in data.items: (try: parseInt(d) except: 0)) == @[0, 0]
+  doAssert collect(for (i, d) in pairs(data): {i: d}) == {1: "word",
+      0: "bird"}.toTable
+  doAssert collect(for d in data.items: {d}) == data.toHashSet
+
```

Comment:
future PR welcome to do the following: https://github.com/timotheecour/Nim/issues/534#issuecomment-769565033
(or in this PR if you can)

Reply:
That actually made me find a bug (#12595) with using `{k: v}` in templates.